File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() <= 1) return nums.size();
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int count = 1;
        int max_count = 1;
        for(int i = 1; i < n; i++){
            if(nums[i-1] == nums[i]){
                continue;
            }
            if(nums[i-1] + 1 == nums[i]){
                count++;
                max_count = max(count, max_count);
            }
            else{
                count = 1;
            }
        }
        return max_count; 
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        n = len(nums)
        if n <= 1:
            return n

        temp = sorted(nums)

        count = 1
        max_count = 0
        for i in range(1, n):
            if temp[i-1] == temp[i]:
                continue
            if temp[i-1] + 1 == temp[i]:
                count += 1
                max_count = max(count, max_count)
            else:
                count = 1
        return max(max_count, count)
